nextflow/src/org/labkey/nextflow/NextFlowController.java

```diff
@@ -0,0 +1,307 @@
+        }
+
+        @Override
+        public boolean handlePost(NextFlowConfiguration nextFlowConfiguration, BindException errors) throws Exception
+        {
+            NextFlowManager.get().addConfiguration(nextFlowConfiguration, errors);
+            return !errors.hasErrors();
+        }
+
+        @Override
+        public URLHelper getSuccessURL(NextFlowConfiguration nextFlowConfiguration)
+        {
+            return getContainer().getStartURL(getUser());
+        }
+
+        @Override
+        public void addNavTrail(NavTree root)
+        {
+
+        }
+    }
+
+    public static class NextFlowConfiguration
+    {
+        private String nextFlowConfigFilePath;
+        private String accountName;
+        private String identity;
+        private String s3BucketPath;
+        private String credential;
+
+        public String getNextFlowConfigFilePath()
+        {
+            return nextFlowConfigFilePath;
+        }
+
+        public void setNextFlowConfigFilePath(String nextFlowConfigFilePath)
+        {
+            this.nextFlowConfigFilePath = nextFlowConfigFilePath;
+        }
+
+        public String getAccountName()
+        {
+            return accountName;
+        }
+
+        public void setAccountName(String accountName)
+        {
+            this.accountName = accountName;
+        }
+
+        public String getIdentity()
+        {
+            return identity;
+        }
+
+        public void setIdentity(String identity)
+        {
+            this.identity = identity;
+        }
+
+        public String getS3BucketPath()
+        {
+            return s3BucketPath;
+        }
+
+        public void setS3BucketPath(String s3BucketPath)
+        {
+            this.s3BucketPath = s3BucketPath;
+        }
+
+        public String getCredential()
+        {
+            return credential;
+        }
+
+        public void setCredential(String credential)
+        {
+            this.credential = credential;
+        }
+    }
+
+    @RequiresPermission(SiteAdminPermission.class)
+    public static class NextFlowEnableAction extends FormViewAction
+    {
+
+        @Override
+        public void validateCommand(Object target, Errors errors)
+        {
+
+        }
+
+        @Override
+        public ModelAndView getView(Object form, boolean reshow, BindException errors) throws Exception
+        {
+            PropertyStore store = PropertyManager.getNormalStore();
+            PropertyManager.PropertyMap map = store.getProperties(NextFlowManager.NEXTFLOW_ENABLE);
+            String btnTxt = "Enable NextFlow";
+            // check if nextflow is enabled
+            if (Boolean.parseBoolean(map.get(IS_NEXTFLOW_ENABLED)))
+            {
+                btnTxt = "Disable NextFlow";
+            }
+            else
+            {
+                btnTxt = "Enable NextFlow";
+            }
+
+            return new HtmlView("Enable/Disable Nextflow", DIV("Enable/Disable Nextflow",
```

Comment:
These kinds of UIs can be confusing about what's the current vs future state.

In addition to the button, let's clearly state "NextFlow is currently enabled in this folder" (or disabled).

Reply:
done